storage/ndb/rest-server/rest-api-server/internal/feature_store/metadata.go

```diff
@@ -196,7 +207,7 @@ func newFeatureViewMetadata(
 	var fgSchemaCache = make(map[int]*dal.PerFeatureAvroSchema)
 	for _, fgFeature := range fgFeaturesArray {
 		for _, feature := range fgFeature.Features {
-			if (*feature).IsComplex() {
+			if (*feature).IsComplex() && !(*feature).IsOnDemand() {
```

Comment:
need to check if the feature group is spine instead.
https://github.com/logicalclocks/hopsworks-ee/blob/master/docker/migration/sql/ddl/V4.0.0__initial_tables.sql#L2017